### PR TITLE
fix(request): replace is-array-buffer module with inline equivalent

### PR DESCRIPTION
**Pull request by wyvern8**
is-array-buffer causes issues with webpack

addresses #774

### CODE REVIEW

File: lib/request/request.js [hunk 1]
```diff
@@ -22,6 +21,22 @@ function httpRequest (origRequestOptions) {
...
+  * Released under the MIT license
+  *
+  * Date: 2017-07-26T11:00:44.931Z
+  */
+  function isArrayBuffer (value) {
+    var hasArrayBuffer = typeof ArrayBuffer === 'function'
+    var toString = Object.prototype.toString
+    return hasArrayBuffer && (value instanceof ArrayBuffer || toString.call(value) === '[object ArrayBuffer]')
+  }
+  // --- end isArrayBuffer
```
Comment (gr2m) [inline lib/request/request.js]: could you please move it into ` lib/request/is-array-buffer.js`, mention that it’s a temporary workaround and reference https://github.com/octokit/rest.js/issues/774

File: .gitignore [hunk 2]
```diff
@@ -19,3 +19,8 @@ dist/
 
 # cypress
 cypress/
+
+# intellij
+.idea/
+*.iml
+
```
Comment (gr2m) [inline .gitignore]: could you please remove this unrelated change?

You can set global gitignore rules in `~/.gitignore` so you don’t need to set it per project :)

Comment (wyvern8) [reply .gitignore]: yeah good point - want to leave it there for others, or remove?

File: lib/request/is-array-buffer.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * Date: 2017-07-26T11:00:44.931Z
+ */
+
+'use strict'
+
+// temporary workaround for https://github.com/octokit/rest.js/issues/774
+
+var hasArrayBuffer = typeof ArrayBuffer === 'function'
+var toString = Object.prototype.toString
```
Comment (gr2m) [inline lib/request/is-array-buffer.js]: can you use `const` please?

File: .gitignore (hunk 2 above)
Comment (gr2m) [reply .gitignore]: please remove, thank you!

Comment (wyvern8) [reply .gitignore]: done

**Review by gr2m — DISMISSED** [review]
thanks :)

**Comment by gr2m** [comment]
I’ll merge & release it together with #776

**Review by gr2m — APPROVED** [review]
